src/lib/validate-processor.ts

```diff
@@ -59,12 +59,10 @@ export async function validateProcessor(
         localResponse.allowed = resp.allowed;
 
         // If the validation callback returned a status code or message, set it in the Response
-        if (resp.statusCode || resp.statusMessage) {
-          localResponse.status = {
-            code: resp.statusCode || 400,
-            message: resp.statusMessage || `Validation failed for ${name}`,
-          };
-        }
+        localResponse.status = {
+          code: resp.statusCode || 400,
+          message: resp.statusMessage || `Validation failed for ${name}`,
+        };
```

Comment:
What happens if `resp` does not include either property but is otherwise okay? Is that possible? We should look into that before merging.